library/psa_util.c

```diff
@@ -0,0 +1,154 @@
+    PSA_ERROR_NOT_SUPPORTED,         MBEDTLS_ERR_MD_FEATURE_UNAVAILABLE,
+    PSA_ERROR_INVALID_ARGUMENT,      MBEDTLS_ERR_MD_BAD_INPUT_DATA,
+    PSA_ERROR_INSUFFICIENT_MEMORY,   MBEDTLS_ERR_MD_ALLOC_FAILED
+};
+#endif
+#if defined(MBEDTLS_LMS_C)
+psa_status_t psa_to_lms_errors[] =
+{
+    PSA_SUCCESS,                     0,
+    PSA_ERROR_BUFFER_TOO_SMALL,      MBEDTLS_ERR_LMS_BUFFER_TOO_SMALL,
+    PSA_ERROR_INVALID_ARGUMENT,      MBEDTLS_ERR_LMS_BAD_INPUT_DATA
+};
+#endif
+#if defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_SSL_PROTO_TLS1_3)
+psa_status_t psa_to_ssl_errors[] =
+{
+    PSA_SUCCESS,                     0,
+    PSA_ERROR_INSUFFICIENT_MEMORY,   MBEDTLS_ERR_SSL_ALLOC_FAILED,
+    PSA_ERROR_NOT_SUPPORTED,         MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE,
+    PSA_ERROR_INVALID_SIGNATURE,     MBEDTLS_ERR_SSL_INVALID_MAC,
+    PSA_ERROR_INVALID_ARGUMENT,      MBEDTLS_ERR_SSL_BAD_INPUT_DATA,
+    PSA_ERROR_BAD_STATE,             MBEDTLS_ERR_SSL_INTERNAL_ERROR,
+    PSA_ERROR_BUFFER_TOO_SMALL,      MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL
+};
+#endif
+
+#if defined(PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY) ||    \
+    defined(PSA_WANT_KEY_TYPE_RSA_KEY_PAIR)
+psa_status_t psa_to_pk_rsa_errors[] =
+{
+    PSA_SUCCESS,                     0,
+    PSA_ERROR_NOT_PERMITTED,         MBEDTLS_ERR_RSA_BAD_INPUT_DATA,
+    PSA_ERROR_INVALID_ARGUMENT,      MBEDTLS_ERR_RSA_BAD_INPUT_DATA,
+    PSA_ERROR_INVALID_HANDLE,        MBEDTLS_ERR_RSA_BAD_INPUT_DATA,
+    PSA_ERROR_BUFFER_TOO_SMALL,      MBEDTLS_ERR_RSA_OUTPUT_TOO_LARGE,
+    PSA_ERROR_INSUFFICIENT_ENTROPY,  MBEDTLS_ERR_RSA_RNG_FAILED,
+    PSA_ERROR_INVALID_SIGNATURE,     MBEDTLS_ERR_RSA_VERIFY_FAILED,
+    PSA_ERROR_INVALID_PADDING,       MBEDTLS_ERR_RSA_INVALID_PADDING
+};
+#endif
+
+#if defined(MBEDTLS_USE_PSA_CRYPTO) && \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY)
+psa_status_t psa_to_pk_ecdsa_errors[] =
+{
+    PSA_SUCCESS,                     0,
+    PSA_ERROR_NOT_PERMITTED,         MBEDTLS_ERR_ECP_BAD_INPUT_DATA,
+    PSA_ERROR_INVALID_ARGUMENT,      MBEDTLS_ERR_ECP_BAD_INPUT_DATA,
+    PSA_ERROR_INVALID_HANDLE,        MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE,
+    PSA_ERROR_BUFFER_TOO_SMALL,      MBEDTLS_ERR_ECP_BUFFER_TOO_SMALL,
+    PSA_ERROR_INSUFFICIENT_ENTROPY,  MBEDTLS_ERR_ECP_RANDOM_FAILED,
+    PSA_ERROR_INVALID_SIGNATURE,     MBEDTLS_ERR_ECP_VERIFY_FAILED
+};
+#endif
+
+int psa_generic_status_to_mbedtls( psa_status_t status )
+{
+    switch( status )
+    {
+    case PSA_SUCCESS:
+        return( 0 );
+    case PSA_ERROR_NOT_SUPPORTED:
+        return( MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED );
+    case PSA_ERROR_CORRUPTION_DETECTED:
+        return( MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED );
+    case PSA_ERROR_COMMUNICATION_FAILURE:
+    case PSA_ERROR_HARDWARE_FAILURE:
+        return( MBEDTLS_ERR_PLATFORM_HW_ACCEL_FAILED );
+    case PSA_ERROR_NOT_PERMITTED:
+    default:
+        return( MBEDTLS_ERR_ERROR_GENERIC_ERROR );
+    }
+}
+
+int psa_status_to_mbedtls( psa_status_t status,
```

Comment:
Perhaps if we add another `enum` parameter here to signal from which module the call is coming from the function might be able to select the proper "translation buffer" on its own. Wdyt?